Source/WebKit/NetworkProcess/storage/LocalStorageManager.cpp

```diff
@@ -198,17 +198,18 @@ StorageAreaIdentifier LocalStorageManager::connectToLocalStorageArea(IPC::Connec
         m_registry->registerStorageArea(storage->identifier(), *storage);
     }
 
-    ASSERT(m_path.isEmpty() || is<SQLiteStorageArea>(*m_localStorageArea));
-    protectedLocalStorageArea()->addListener(connection, sourceIdentifier);
-    return m_localStorageArea->identifier();
+    return m_localStorageArea.get();
 }
 
-RefPtr<StorageAreaBase> LocalStorageManager::protectedLocalStorageArea() const
+StorageAreaIdentifier LocalStorageManager::connectToLocalStorageArea(IPC::Connection::UniqueID connection, StorageAreaMapIdentifier sourceIdentifier, const WebCore::ClientOrigin& origin, Ref<WorkQueue>&& workQueue)
 {
-    return m_localStorageArea;
+    RefPtr localStorageArea = ensureLocalStorageArea(origin, WTFMove(workQueue));
+    ASSERT(m_path.isEmpty() || is<SQLiteStorageArea>(*localStorageArea));
+    localStorageArea->addListener(connection, sourceIdentifier);
+    return localStorageArea->identifier();
 }
 
-StorageAreaIdentifier LocalStorageManager::connectToTransientLocalStorageArea(IPC::Connection::UniqueID connection, StorageAreaMapIdentifier sourceIdentifier, const WebCore::ClientOrigin& origin)
+MemoryStorageArea* LocalStorageManager::ensureTransientLocalStorageArea(const WebCore::ClientOrigin& origin)
```

Comment:
Why is this returning a raw pointer and not a C++ reference? Seems like it can never return null?